pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -618,3 +615,79 @@ def _log_B_three(i):
         return purchase_prob.transpose(
             "chain", "draw", "customer_id", missing_dims="ignore"
         )
+
+    def _population_distributions(
+        self,
+        random_seed: Optional[RandomState] = None,
+        var_names: Sequence[str] = ("population_dropout", "population_purchase_rate"),
+    ) -> xarray.Dataset:
+        """Utility function for posterior predictive sampling from dropout and purchase rate distributions."""
+        with pm.Model():
+            # purchase rate priors
+            r = pm.HalfFlat("r")
+            alpha = pm.HalfFlat("alpha")
+
+            # dropout priors
+            s = pm.HalfFlat("s")
+            beta = pm.HalfFlat("beta")
+
+            # This is the shape if using fit_method="map"
+            if self.fit_result.dims == {"chain": 1, "draw": 1}:
+                shape_kwargs = {"shape": 4000}
+            else:
+                shape_kwargs = {}
+
+            pm.Gamma("population_purchase_rate", mu=r, sigma=alpha, **shape_kwargs)
+            pm.Gamma("population_dropout", mu=s, sigma=beta, **shape_kwargs)
```

Comment:
Are you sure those parameters correspond to `mu` and `sigma`. Statistical analysis of the `Gamma` usually work with `rate` and `shape`, so I would expect that's what the Pareto marginalized likelihood is defined in terms of

Reply:
The research papers all state the `scale` parameter is used instead of `rate`, but I don't see `scale` among the arguments for `pm.Gamma`.

Using the `rate` parameter results in distributions like this, which makes no sense for a dataset where most transactions are days apart:

![image](https://github.com/pymc-labs/pymc-marketing/assets/10178857/74f017e8-13f4-4609-bea3-eb7fc573569c)

Reply:
`scale = 1 / rate`. Does that seem to work? We should confirm carefully though.

Reply:
It seems to, after running it through this PDF app:

https://homepage.divms.uiowa.edu/~mbognar/applets/gamma.html

Reply:
I meant to confirm this is the parametrization that the paper uses. But given the raw outputs I am pretty convinced it is

Reply:
From https://www.brucehardie.com/notes/009/pareto_nbd_derivations_2005-11-05.pdf:

![Screenshot 2023-07-16 at 12 42 37 PM](https://github.com/pymc-labs/pymc-marketing/assets/10178857/c50fd565-4298-4905-9d1e-bc3e1d969664)

Reply:
No DIV/0 warnings came up when I tested this with NUTS in the notebook, so I don't see a need to suppress those warnings for now.